pkg/pgsql/server/fmessages/bind.go

```diff
@@ -0,0 +1,154 @@
+	ResultColumnFormatCodes []int16
+}
+
+func ParseBindMsg(payload []byte) (BindMsg, error) {
+	b := bytes.NewBuffer(payload)
+	r := bufio.NewReader(b)
+	destPortalName, err := getNextString(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	preparedStatement, err := getNextString(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	// The number of parameter format codes that follow (denoted C below).
+	// This can be zero to indicate that there are no parameters or that the parameters all use the default format (text);
+	// or one, in which case the specified format code is applied to all parameters; or it can equal the actual number
+	// of parameters.
+	parameterFormatCodeNumber, err := getNextInt16(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	parameterFormatCodes := make([]int16, parameterFormatCodeNumber)
+	for k := 0; k < int(parameterFormatCodeNumber); k++ {
+		p, err := getNextInt16(r)
+		if err != nil {
+			return BindMsg{}, err
+		}
+		parameterFormatCodes[k] = p
+	}
+	// The number of parameter values that follow (possibly zero). This must match the number of parameters needed by the query.
+	pCount, err := getNextInt16(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+
+	// Handling format codes: see resultColumnFormatCodesNumber property comment
+	forceTXT := false
+	forceBIN := false
+	if len(parameterFormatCodes) == 0 {
+		forceTXT = true
+	}
+	if len(parameterFormatCodes) == 1 {
+		switch parameterFormatCodes[0] {
+		case 0:
+			forceTXT = true
+		case 1:
+			forceBIN = true
+		default:
+			return BindMsg{}, errors.New("malformed bind message. Allowed format codes are 1 or 0")
+		}
+	}
+
+	if len(parameterFormatCodes) > 1 && len(parameterFormatCodes) != int(pCount) {
+		return BindMsg{}, errors.New("malformed bind message. Parameters format codes didn't match parameters count")
+	}
+
+	params := make([]interface{}, 0)
+	for i := 0; i < int(pCount); i++ {
+		pLen, err := getNextInt32(r)
+		if err != nil {
+			return BindMsg{}, err
+		}
+		pVal := make([]byte, pLen)
+		_, err = r.Read(pVal)
+		if err != nil {
+			return BindMsg{}, err
+		}
+		if forceTXT {
+			params = append(params, string(pVal))
+			continue
+		}
+		if forceBIN {
+			params = append(params, pVal)
+			continue
+		}
+
+		switch parameterFormatCodes[i] {
+		case 0:
+			params = append(params, string(pVal))
+		case 1:
+			params = append(params, pVal)
+		}
+	}
+	// The number of result-column format codes that follow (denoted R below).
+	// This can be zero to indicate that there are no result columns or that the result columns should all use the
+	// default format (text); or one, in which case the specified format code is applied to all result columns (if any);
+	// or it can equal the actual number of result columns of the query.
+	resultColumnFormatCodesNumber, err := getNextInt16(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+
+	resultColumnFormatCodes := make([]int16, 0, resultColumnFormatCodesNumber)
```

Comment:
`resultColumnFormatCodesNumber` can be negative here causing a panic. Please sanitize it.